src/script/hostgraph/build_hostgraph.sh

```diff
@@ -178,68 +185,101 @@ MERGE_CRAWL_INPUT=""
 for CRAWL in ${CRAWLS[@]}; do
 
     INPUT_SPLITS=($(create_input_splits $CRAWL))
-    echo "Input splits: ""${INPUT_SPLITS[*]}"
-
-    for ((i=0; i<${#INPUT_SPLITS[@]}; i++)); do
-        INPUT=${INPUT_SPLITS[$i]}
-        NUM_INPUT_PATHS=$(wc -l <input/$CRAWL/$(basename $INPUT))
-        INPUT_PARTITIONS=$((NUM_INPUT_PATHS/DIVISOR_INPUT_PARTITIONS))
-        echo "$INPUT => $INPUT_PARTITIONS partitions"
-
-        _step hostlinks.$CRAWL.split$i \
-              "$SPARK_HOME"/bin/spark-submit \
-              $SPARK_ON_YARN \
-              $SPARK_HADOOP_OPTS \
-              --conf spark.serializer=org.apache.spark.serializer.KryoSerializer \
-              --conf spark.task.maxFailures=80 \
-              --conf spark.executor.memory=$EXECUTOR_MEM \
-              --conf spark.driver.memory=6g \
-              --conf spark.core.connection.ack.wait.timeout=600s \
-              --conf spark.network.timeout=300s \
-              --conf spark.shuffle.io.maxRetries=5 \
-              --conf spark.shuffle.io.retryWait=30s \
-              --conf spark.io.compression.codec=zstd \
-              --conf spark.checkpoint.compress=true \
-              --conf spark.locality.wait=0s \
-              --num-executors $NUM_EXECUTORS \
-              --executor-cores $EXECUTOR_CORES \
-              --executor-memory $EXECUTOR_MEM \
-              --conf spark.sql.warehouse.dir=$WAREHOUSE_DIR/$CRAWL \
-              --conf spark.sql.parquet.compression.codec=zstd \
-              --py-files "$PYFILES_HOST_LINK_EXTRACTOR" \
-              $SPARK_EXTRA_OPTS \
-              $HOST_LINK_EXTRACTOR \
-              --input_base_url $INPUT_BASE_URL \
-              --num_input_partitions $INPUT_PARTITIONS \
-              --num_output_partitions $OUTPUT_PARTITIONS \
-              --local_temp_dir "$TMPDIR" \
-              $INPUT hostlinks$i
-        #         --intermediate_output hostlinkstmp$i
-
-        _step hostlinks.$CRAWL.split$i.distcp \
-              hadoop distcp \
-              -Dfs.s3a.connection.timeout=2000 \
-              -Dfs.s3a.attempts.maximum=3 \
-              "$HDFS_BASE_DIR"/$CRAWL/hostlinks$i \
-              "$S3A_OUTPUT_PREFIX"/$CRAWL/hostlinks/$i
-
-    done
 
+    if [ -z "$INPUT_SPLITS" ]; then
+        # no input splits signals that the crawl has already successfully processed
+        if hadoop fs -stat "$S3A_OUTPUT_PREFIX"/$CRAWL/hostlinks/; then
+            echo "Output prefix for crawl $CRAWL already exists on S3: $S3A_OUTPUT_PREFIX/$CRAWL/hostlinks/"
+            if ! hadoop fs -stat "$S3A_OUTPUT_PREFIX"/$CRAWL/hostlinks/_SUCCESS; then
+                echo "No success marker found below S3 output prefix: $S3A_OUTPUT_PREFIX/$CRAWL/hostlinks/_SUCCESS"
+                echo "Please, verify the output and depending on the verification result, manually add the success marker or remove the output. Exiting ..."
```

Comment:
Is there a role requirement for manually creating these `_SUCCESS` markers?

Reply:
The script copies the data of the output splits into the same folder / prefix. It would fail earlier if it had no write access to this bucket prefix.